### ISSUE TITLE
Introduce 'consistent' value for 'array-element-newline'.

### ISSUE THREAD

**Issue by vuchastyi** [eslint/eslint#10310]
**What rule do you want to change?**
[`array-element-newline`](https://eslint.org/docs/rules/array-element-newline)
**Does this change cause the rule to produce more or fewer warnings?**
if I correctly understood the question - then yes. In my situation it cause very much errors, which i can't change anything, except change my code style - which is not prefered.
**How will the change be implemented? (New option, new default behavior, etc.)?**
New value: `'array-element-newline':  ['error', 'consistent']`
**Please provide some example code that this change will affect:**

```js
const d = {
  'padding-line-between-statements': [
      error,
      { blankLine: always, prev: 'directive', next: '*' },
      { blankLine: 'any', prev: 'directive', next: 'directive' },
      { blankLine: always, prev: ['const', 'let', 'var'], next: '*' },
      { blankLine: 'any', prev: ['const', 'let', 'var'], next: ['const', 'let', 'var'] },
      { blankLine: 'always', prev: '*', next: 'return' }
    ]
};
```

**What does the rule currently do for this code?**
Currently rule throws an error, which i can't fix in any way.
**What will the rule do after it's changed?**
It will pass without any errors 😄

**Comment by platinumazure** [comment]
Hi @vuchastyi, thanks for the issue.

The second question is about what effect the new rule behavior would have-- whether it would cause more warnings than current behavior, fewer warnings than current behavior, or either way because it's situational. In this case, the "consistent" option would generate more warnings in some cases and fewer warnings in others. Hope this helps.

Thanks for filling out the rule change proposal template, it really helps us out!

**Comment by vuchastyi** [comment]
Hey @platinumazure!
Sorry, my english lil bad, so i didn't understand what you wrote.
What i need to do?
And i don't understand what i need to do with this issue. Do i need to close it and maintainers later will create PRs for this issue or i need to leave it opened?
Thanks for your answer!
Regards, Igor Vuchastyi.

**Comment by vuchastyi** [comment]
<details>
<summary>Some screenshots from real life:</summary>

![image](https://user-images.githubusercontent.com/19193608/39670637-ed7b9af0-5111-11e8-828a-41945777b898.png)
![image](https://user-images.githubusercontent.com/19193608/39670642-fd31e472-5111-11e8-9900-93df2439d5f9.png)
</details>

**Comment by revolter** [comment]
Wasn't this fixed by #10355?

**Comment by not-an-aardvark** [comment]
Yes, it looks like it was. Closing this issue.
